dbms/src/IO/Compression/CompressedReadBufferFromFile.cpp

```diff
@@ -12,28 +12,36 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <IO/Buffer/createReadBufferFromFileBase.h>
 #include <IO/Compression/CompressedReadBufferFromFile.h>
-#include <IO/Util/WriteHelpers.h>
-
 
 namespace DB
 {
+
 namespace ErrorCodes
 {
 extern const int SEEK_POSITION_OUT_OF_BOUND;
 }
 
 template <bool has_legacy_checksum>
-bool CompressedReadBufferFromFile<has_legacy_checksum>::nextImpl()
+CompressedReadBufferFromFileImpl<has_legacy_checksum>::CompressedReadBufferFromFileImpl(
+    std::unique_ptr<ReadBufferFromFileBase> && file_in_)
+    : CompressedSeekableReaderBuffer()
+    , p_file_in(std::move(file_in_))
+    , file_in(*p_file_in)
+{
+    this->compressed_in = &file_in;
+}
+
+template <bool has_legacy_checksum>
+bool CompressedReadBufferFromFileImpl<has_legacy_checksum>::nextImpl()
 {
     size_t size_decompressed;
     size_t size_compressed_without_checksum;
     size_compressed = this->readCompressedData(size_decompressed, size_compressed_without_checksum);
     if (!size_compressed)
         return false;
 
-    assert(size_decompressed > 0 && size_compressed_without_checksum > 0);
```

Comment:
why `size_compressed_without_checksum > 0` is removed?

Reply:
Not necessary, this file is rename from `CompressedReadBufferFromFileProvider.cpp`

The assert is used to pass tidy, `size_compressed_without_checksum > 0` is not necessary.